tests/model_hub_tests/pytorch/test_stateful_to_stateless_transformation.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import openvino as ov
+from openvino._offline_transformations import stateful_to_stateless_transformation
+from optimum.intel import OVModelForCausalLM
+import models_hub_common.utils as utils
+import pytest
+import os
+
+def get_read_value_ops(model: ov.Model):
+    return [op for op in model.get_ops() if op.get_type_name() == 'ReadValue']
+
+def check_desc_tensors(tensors1, tensors2):
+    # order of tensors may not match, comparing by the total amount and names
+    assert len(tensors1) == len(tensors2)
+    assert set(tuple(t.names) for t in tensors1) == set(tuple(t.names) for t in tensors2)
+    for t1 in tensors1:
+        t2_candidates = [t for t in tensors2 if t1.names & t.names]
+        assert len(t2_candidates) == 1
+        t2 = t2_candidates[0]
+        assert t1.names == t2.names
+        assert t1.get_partial_shape() == t2.get_partial_shape()
+        assert t1.get_element_type() == t2.get_element_type()
+
+def run_stateful_to_stateless_in_runtime(tmp_path, model_id, model_link):
+    model = OVModelForCausalLM.from_pretrained(model_id, export=True, stateful=True, compile=False)
+    assert len(model.model.get_sinks()), f"Input model is not in the expected stateful form because it doesn't have any sinks."
+    assert len(get_read_value_ops(model.model)), f"Input model is not in the expected stateful form because it doesn't have any ReadValue operations."
```

Comment:
I usually set `trust_remote_code=True` in `from_pretrained()` methods as it is mandatory for some reason. Don't know if this is required for some models. Or prove I'm doing wrong :)

Reply:
It is based on your level of trust. Some things push my trust level low when it comes to model conversion from particular sources. Just joking.

Reply:
Now, when this PR is merged, @CuriousPanCake, you can check if `trust_remote_code=True` works with this transformation ;)